pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -127,20 +127,23 @@ func (r *Reconciler) checkQuota(guestAccelerators []machinev1.GCPGPUConfig) erro
 }
 
 func (r *Reconciler) validateGuestAccelerators() error {
-	// Note(elmiko) this is known to have an error in that non a2 instances with GPUs (eg a3 types) will bypass this check, which is fine for now.
-	if len(r.providerSpec.GPUs) == 0 && !strings.HasPrefix(r.providerSpec.MachineType, "a2-") {
+	// Note, this logic will reject machines that might are not a2 or a3, or have GPUs specified. This means that future
```

Comment:
> Note, this logic will reject machines that might are not a2 or a3

Not exactly sure what this is meant to say. Maybe omitting `might`?

Reply:
oh yeah, good catch